tools/cockpit.spec

```diff
@@ -82,18 +82,54 @@ BuildRequires: krb5-server
 # For documentation
 BuildRequires: xmlto
 
+# == Mandatory components of cockpit == #
+
+# Nothing will work without these sub-packages
 Requires: %{name}-bridge = %{version}-%{release}
-Requires: %{name}-networkmanager = %{version}-%{release}
-Requires: %{name}-ws = %{version}-%{release}
 Requires: %{name}-shell = %{version}-%{release}
-Requires: %{name}-storaged = %{version}-%{release}
-%ifarch x86_64 armv7hl
-Requires: %{name}-docker = %{version}-%{release}
-%endif
+
 %if 0%{?rhel}
 Requires: %{name}-subscriptions = %{version}-%{release}
 %endif
 
+# == Optional components == #
+
+# Fedora 24+ supports "soft" dependencies
+%if 0%{?fedora} >= 24 || 0%{?rhel} >= 8
+
+# Recommends: are components that *should* be installed
+# by default, but may be omitted if desired. Removing
+# a Recommended sub-package from the system will not
+# remove the cockpit meta-package.
+Recommends: %{name}-ws = %{version}-%{release}
```

Comment:
OK, I was a little unsure about this one. I wasn't sure if we'd want to consider "Cockpit" still installed if it was only the "headless" version. If the preference is for the metapackage to explicitly require `cockpit-ws`, I'm perfectly fine with that.